apis/python/src/tiledbsoma/soma_dataframe.cc

```diff
@@ -156,6 +159,57 @@ void load_soma_dataframe(py::module& m) {
             },
             "column_names"_a)
 
+        .def(
+            "extend_enumeration_values",
+            [](SOMADataFrame& sdf,
+               // Map values are of type pa.Array.
+               //
+               // One might think this looks a lot like a pa.RecordBatch,
+               // so why not use that? Answer: the map values are, in general,
+               // all of different lengths. So they cannot be formed into
+               // an Arrow table and passed into here that way.
+               std::map<std::string, py::object> values,
+               bool deduplicate) -> py::none {
+                size_t ncol = values.size();
+                std::vector<ArrowSchema> arrow_schemas(ncol);
+                std::vector<ArrowArray> arrow_arrays(ncol);
+                size_t i = 0;
+                std::map<std::string, std::pair<ArrowSchema*, ArrowArray*>>
+                    map_for_cpp;
+                for (auto item : values) {
+                    std::string column_name = py::str(item.first);
+                    py::object pa_array_for_column = item.second;
+                    uintptr_t arrow_schema_ptr = (uintptr_t)(&arrow_schemas[i]);
+                    uintptr_t arrow_array_ptr = (uintptr_t)(&arrow_arrays[i]);
```

Comment:
Nit: Avoid c-style casting. Named casts provided more context for future developers and have fewer foot-guns.

Reply:
For this spot:

```
error: static_cast from 'value_type *' (aka 'ArrowSchema *') to 'uintptr_t' (aka 'unsigned long') is not allowed
  182 |                     uintptr_t arrow_schema_ptr = static_cast<uintptr_t>(&arrow_schemas[i]);
      |                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

:(